source/s3_auto_ranged_put.c

```diff
@@ -1220,6 +1010,27 @@ static void s_s3_prepare_upload_part_on_read_done(void *user_data) {
         aws_s3_meta_request_unlock_synced_data(meta_request);
     }
     /* END CRITICAL SECTION */
+    if (uploaded_part_info) {
+        /* Check for the durability */
+        if (request->request_body.capacity != uploaded_part_info->size) {
```

Comment:
it's weird to compare capacity, when you could just use length
```suggestion
        if (request->request_body.len != uploaded_part_info->size) {
```